### PR TITLE
Add copy code button to modal

### PR DESCRIPTION
**Pull request by mikemcbride**
Adds a copy code button to the box-shadow code in the modal:

https://user-images.githubusercontent.com/6248612/193976649-0b803cd6-fac8-4387-8571-9b620f7d4022.mov

@davidleininger any feedback on this one?

### CODE REVIEW

File: src/components/CodeContainer.vue [hunk 1]
```diff
@@ -1,18 +1,26 @@
 <template>
     <pre
-      class="p-4 mb-6 font-mono text-gray-800 bg-gray-50 border border-gray-200 rounded-md overflow-x-scroll"
-    ><code :class="languageClass">{{ content }}</code></pre>
+      class="p-4 mb-6 font-mono text-gray-800 bg-gray-50 border border-gray-200 rounded-md overflow-x-scroll relative flex items-center"
+    ><code class="flex-1" :class="languageClass">{{ content }}</code>
+        <span v-if="showCopiedText" class="bg-gray-50 px-4 font-semibold text-emerald-500 text-sm font-sans absolute right-0">Copied!</span>
+        <button v-else @click="copyToClipboard" title="Copy to clipboard" class="bg-gray-50 text-gray-400 hover:text-gray-800 p-2 focus:outline-none focus:ring focus:ring-blue-600 rounded absolute right-1 inline-flex">
```
Comment (davidleininger) [inline src/components/CodeContainer.vue:6]:
```suggestion
        <button v-else @click="copyToClipboard" title="Copy to clipboard" class="bg-gray-50 text-gray-500 hover:text-blue-600 p-2 focus:outline-none focus:ring focus:ring-blue-600 rounded absolute right-1 inline-flex">
```

File: src/components/CodeContainer.vue [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+          <ClipboardIcon class="h-5 w-5" />
```
Comment (davidleininger) [inline src/components/CodeContainer.vue:7]:
```suggestion
          <ClipboardIcon class="h-5 w-5" /> copy
```

**Review by davidleininger — CHANGES_REQUESTED** [review]
image

The copy button is inset a little and the text goes behind it. Personally, I'd like to see it as a separate button that is just inside. This way you won't see content behind it. It would also be nice to add the text `copy` next to the item. This will give an an accessible name. That said these are MINOR changes and I think we could merge as is and then go back to it.

Lastly, the copy button doesn't pass contrast accessibility. I would make a few changes, I'll include as a suggestion. Also, the main text in the code element passes contrast, but really only for UI elements, not text, I'd bump it to match `text-gray-500`

**Comment by mikemcbride** [comment]
@davidleininger let me know what you think of these changes. Screenshot for posterity:

image

**Review by davidleininger — APPROVED** [review]
Ship it.

**Pull request merged**